### ISSUE TITLE
Allow group names with `-`

### ISSUE THREAD

**Issue by camilamacedo86** [kubernetes-sigs/kubebuilder#954]
## Currently:

```
 $ kubebuilder create api --group webapp-hyphen --version v1 --kind Guestbook
Create Resource [y/n]
y
Create Controller [y/n]
y
Writing scaffold for you to edit...
api/v1/guestbook_types.go
2019/08/15 23:10:22 error scaffolding APIs: group must match ^[a-z]+$ (was webapp-hyphen)
```

## RFE
However, the controller allows it. The change is just in the regex ^[a-z]+$.

**Comment by camilamacedo86** [comment]
PR: https://github.com/kubernetes-sigs/kubebuilder/pull/977

**Comment by dlespiau** [comment]
It seems this issue should be closed now that #977 is merged.

**Comment by dlespiau** [comment]
Thank you!
